src/SignalR/common/Shared/ReflectionHelper.cs

```diff
@@ -47,26 +47,28 @@ public static bool TryGetStreamType(Type streamType, [NotNullWhen(true)] out Typ
         return false;
     }
 
-    public static bool IsIAsyncEnumerable([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)] Type type)
```

Comment:
The annotation wasn't "wrong" necessarily. The problem was that callers of this API can't guarantee statically which Types are being passed in - they use `someObject.GetType()` to obtain the type. So they were getting warnings they can't address.

This method doesn't need _all_ interfaces on the Type. It just needs to check for 1 interface - `IAsyncEnumerable`. And so referencing that interface directly solves the trimming concern. The trimmer can't remove this interface, because it is being referenced directly.